src/portable/espressif/esp32s2/dcd_esp32s2.c

```diff
@@ -671,16 +755,24 @@ static void _dcd_int_handler(void* arg)
     dcd_event_bus_signal(0, DCD_EVENT_BUS_RESET, true);
   }
 
-  if (int_status & USB_OTGINT_M)
-  {
-    // OTG INT bit is read-only
-    ESP_EARLY_LOGV(TAG, "dcd_int_handler - disconnected");
+  if (int_status & USB_USBSUSP_M) {
+    // start of reset
+    ESP_EARLY_LOGV(TAG, "dcd_int_handler - suspended");
+    USB0.gintsts = USB_USBSUSP_M;
+    //dcd_event_bus_signal(0, DCD_EVENT_SUSPEND, true);
+    dcd_event_bus_signal(0, DCD_EVENT_UNPLUGGED, true);
```

Comment:
suspend is not unplugged, this should be DCD_EVENT_SUSPEND

Reply:
it is in our case.

Reply:
it is a side effect, try to put your PC into sleep to see if the event comes up ?